From 0cb655721d7f8aa5d6d14fc1dfce18c518ce2419 Mon Sep 17 00:00:00 2001
From: Andrew Wedgbury <wedge@sconemad.com>
Date: Mon, 19 Aug 2019 21:12:59 +0000
Subject: [PATCH] CMake-ify

[Retrieve (and updated to only keep fix on sconex.cpp) from:
https://github.com/sconemad/sconeserver/commit/0cb655721d7f8aa5d6d14fc1dfce18c518ce2419]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 .gitignore                          |   5 +
 CMakeLists.txt                      |  52 ++++
 Makefile.am                         |  58 ----
 autogen.sh                          |   8 -
 bluetooth/CMakeLists.txt            |  16 +
 bluetooth/Makefile.am               |  22 --
 config.h.in                         | 335 +--------------------
 configure.in                        | 451 ----------------------------
 debian/rules                        |  10 +-
 debian/sconeserver-base.install     |   2 +-
 debian/sconeserver-dev.install      |   4 +-
 examples/Makefile.am                |   4 -
 examples/rot13/CMakeLists.txt       |   9 +
 examples/rot13/Makefile.am          |  24 --
 examples/tuesdayonly/CMakeLists.txt |   7 +
 examples/tuesdayonly/Makefile.am    |  18 --
 exec/CMakeLists.txt                 |   9 +
 exec/Makefile.am                    |  24 --
 forward/CMakeLists.txt              |   8 +
 forward/Makefile.am                 |  19 --
 http/CMakeLists.txt                 |  51 ++++
 http/Makefile.am                    |  58 ----
 image/CMakeLists.txt                |  12 +
 image/Makefile.am                   |  20 --
 ip/CMakeLists.txt                   |   9 +
 ip/Makefile.am                      |  21 --
 ip6/CMakeLists.txt                  |   9 +
 ip6/Makefile.am                     |  21 --
 local/CMakeLists.txt                |   9 +
 local/Makefile.am                   |  21 --
 location/CMakeLists.txt             |  11 +
 location/Makefile.am                |  18 --
 main.cpp                            |   4 +-
 markdown/CMakeLists.txt             |  20 ++
 markdown/Makefile.am                |  21 --
 maths/CMakeLists.txt                |  32 ++
 maths/Makefile.am                   |  33 --
 mime/CMakeLists.txt                 |   8 +
 mime/Makefile.am                    |  19 --
 mysql/CMakeLists.txt                |  16 +
 mysql/Makefile.am                   |  25 --
 rss/CMakeLists.txt                  |  15 +
 rss/Makefile.am                     |  23 --
 sconeserver.cmake                   |  31 ++
 sconesite/CMakeLists.txt            |  42 +++
 sconesite/Makefile.am               |  79 -----
 sconex/Base64.cpp                   |   2 +-
 sconex/CMakeLists.txt               |  97 ++++++
 sconex/Database.cpp                 |   2 +-
 sconex/FileDir.h                    |   4 +-
 sconex/Makefile.am                  | 156 ----------
 sconex/ModuleLoader.cpp             |   2 +-
 sconex/Process.cpp                  |   8 -
 sconex/RegExp.cpp                   |   4 -
 sconex/RegExp.h                     |   4 +-
 sconex/ScriptContext.cpp            |   7 +-
 sconex/ScriptTypes.cpp              |   2 +-
 sconex/TermBuffer.cpp               |   7 +-
 sconex/sconex.cpp                   |   4 +-
 sconex/sconex.h                     | 176 +++--------
 sconex/utils.cpp                    |   4 -
 server/CMakeLists.txt               |  19 ++
 server/Makefile.am                  |  26 --
 simple/CMakeLists.txt               |  15 +
 simple/Makefile.am                  |  21 --
 smtp/CMakeLists.txt                 |  15 +
 smtp/Makefile.am                    |  21 --
 sqlite/CMakeLists.txt               |  23 ++
 sqlite/Makefile.am                  |  25 --
 ssl/CMakeLists.txt                  |  23 ++
 ssl/Makefile.am                     |  30 --
 stat/CMakeLists.txt                 |  17 ++
 stat/Makefile.am                    |  23 --
 test/CMakeLists.txt                 |  13 +
 test/Makefile.am                    |  53 ----
 testbuilder/CMakeLists.txt          |  23 ++
 testbuilder/Makefile.am             |  36 ---
 tftp/CMakeLists.txt                 |  17 ++
 tftp/Makefile.am                    |  23 --
 upstart                             |  13 -
 80 files changed, 695 insertions(+), 1933 deletions(-)
 create mode 100644 CMakeLists.txt
 delete mode 100644 Makefile.am
 delete mode 100755 autogen.sh
 create mode 100644 bluetooth/CMakeLists.txt
 delete mode 100644 bluetooth/Makefile.am
 delete mode 100644 configure.in
 delete mode 100644 examples/Makefile.am
 create mode 100644 examples/rot13/CMakeLists.txt
 delete mode 100644 examples/rot13/Makefile.am
 create mode 100644 examples/tuesdayonly/CMakeLists.txt
 delete mode 100644 examples/tuesdayonly/Makefile.am
 create mode 100644 exec/CMakeLists.txt
 delete mode 100644 exec/Makefile.am
 create mode 100644 forward/CMakeLists.txt
 delete mode 100644 forward/Makefile.am
 create mode 100644 http/CMakeLists.txt
 delete mode 100644 http/Makefile.am
 create mode 100644 image/CMakeLists.txt
 delete mode 100644 image/Makefile.am
 create mode 100644 ip/CMakeLists.txt
 delete mode 100644 ip/Makefile.am
 create mode 100644 ip6/CMakeLists.txt
 delete mode 100644 ip6/Makefile.am
 create mode 100644 local/CMakeLists.txt
 delete mode 100644 local/Makefile.am
 create mode 100644 location/CMakeLists.txt
 delete mode 100644 location/Makefile.am
 create mode 100644 markdown/CMakeLists.txt
 delete mode 100644 markdown/Makefile.am
 create mode 100644 maths/CMakeLists.txt
 delete mode 100644 maths/Makefile.am
 create mode 100644 mime/CMakeLists.txt
 delete mode 100644 mime/Makefile.am
 create mode 100644 mysql/CMakeLists.txt
 delete mode 100644 mysql/Makefile.am
 create mode 100644 rss/CMakeLists.txt
 delete mode 100644 rss/Makefile.am
 create mode 100644 sconeserver.cmake
 create mode 100644 sconesite/CMakeLists.txt
 delete mode 100644 sconesite/Makefile.am
 create mode 100644 sconex/CMakeLists.txt
 delete mode 100644 sconex/Makefile.am
 create mode 100644 server/CMakeLists.txt
 delete mode 100644 server/Makefile.am
 create mode 100644 simple/CMakeLists.txt
 delete mode 100644 simple/Makefile.am
 create mode 100644 smtp/CMakeLists.txt
 delete mode 100644 smtp/Makefile.am
 create mode 100644 sqlite/CMakeLists.txt
 delete mode 100644 sqlite/Makefile.am
 create mode 100644 ssl/CMakeLists.txt
 delete mode 100644 ssl/Makefile.am
 create mode 100644 stat/CMakeLists.txt
 delete mode 100644 stat/Makefile.am
 create mode 100644 test/CMakeLists.txt
 delete mode 100644 test/Makefile.am
 create mode 100644 testbuilder/CMakeLists.txt
 delete mode 100644 testbuilder/Makefile.am
 create mode 100644 tftp/CMakeLists.txt
 delete mode 100644 tftp/Makefile.am
 delete mode 100644 upstart
 
diff --git a/sconex/sconex.cpp b/sconex/sconex.cpp
index 1b9fc97..a0ba8c1 100644
--- a/sconex/sconex.cpp
+++ b/sconex/sconex.cpp
@@ -35,7 +35,7 @@ VersionTag& version()
 const std::string& sconex_copyright()
 {
   static std::string s_copyright = 
-    "Copyright (c) 2000-2017 Andrew Wedgbury <wedge@sconemad.com>";
+    "Copyright (c) 2000-2018 Andrew Wedgbury <wedge@sconemad.com>";
   return s_copyright;
 }
 
@@ -49,7 +49,7 @@ const std::string& build_type()
 //=========================================================================
 Date& build_time()
 {
-  static Date s_build_time(__DATE__" "__TIME__,true);
+  static Date s_build_time( __DATE__ " " __TIME__ ,true);
   return s_build_time;
 }
